### ISSUE TITLE
whitespace rule enhancements

### ISSUE THREAD

**Issue by rcchen** [palantir/tslint#628]
Create some sort of rule for maximum spacing between characters in a single line I think. For example,

``` ts
public foo { // one space, all good

public foo  { // two spaces, this should be flagged
```

**Comment by adidahiya** [comment]
This should be an extension of the current `whitespace` rule.

**Comment by glen-84** [comment]
@adidahiya It's not only for braces. Right now this is valid:

```js
        if     (1)     {
            let x;
        }    else                 {
            let x;
        }
```

That looks quite awful. Perhaps each `check-*` should have a `spaces` integer which controls the number of spaces that must be used.

However, I would imagine that in 99% of cases, a single space is used/required, so another option would be to just add a `single-spaces` "global" option to enforce a single space in all cases except *before* an equals sign (as people often align assignments).

In a perfect world, it would be able to keep track of groups of assignments and report on misalignment, but that may not be simple.

**Comment by myitcv** [comment]
@rcchen @glen-84 rather than enforcing spacing rules (indeed formatting rules in general) by hand, you might be interested in [`tsfmt`](https://github.com/vvakame/typescript-formatter). It automatically formats code for you.

**Comment by unional** [comment]
Another enhancement: `anonymous-function`
```ts
function() {}

// vs
function () {}
```

**Comment by unional** [comment]
Another enhancement: `anonymous-function`
```ts
function() {}

// vs
function () {}
```

**Comment by glen-84** [comment]
@myitcv Thanks for the suggestion, but I'd prefer to use linting in this case.

**Comment by JoshuaKGoldberg** [comment]
Closing per #4534. Thanks for the discussion folks!
